plugin.rb

```diff
@@ -1,13 +1,52 @@
 # frozen_string_literal: true
 
-# name: discourse-plugin-name
-# about: TODO
+# name: discourse-google-one-tap
+# about: Something
 # version: 0.0.1
-# authors: Discourse
-# url: TODO
+# authors: Ghassan Maslamani
+# url: https://github.com/ghassanmas/discourse-google-one-tap
 # required_version: 2.7.0
 
-enabled_site_setting :plugin_name_enabled
+enabled_site_setting :google_one_tap_enabled
+
+gem "google-id-token", '1.4.2'
+
+require_relative "lib/google_one_tap"
+require_relative "lib/google_one_tap_authenticator"
+register_asset "stylesheets/google_one_tap.scss"
+
+extend_content_security_policy(
+script_src: ['https://accounts.google.com'],
+)
 
 after_initialize do
+
+  register_html_builder('server:before-head-close') do
+    "<script src='https://accounts.google.com/gsi/client' async defer></script>"
+  end
+
+  register_html_builder('server:before-body-close') do |ctx|
+    #This return the div Google JS (loaded above will use)
+    #This could also be done in pure JS
+    #Ref: https://developers.google.com/identity/gsi/web/reference/js-reference
+    # Admin needs to be *really catious* when chaning ifram/container style qouting Google:
+    #> Warning: Do not hide any One Tap prompt content.
+    #> Do not obscure the perception that the One Tap prompt content is from a Google iframe.
+    #> Failure to do so may result in project suspension, account suspension, or both.
+    # Ref https://developers.google.com/identity/gsi/web/guides/change-position
+    current_hostname = Discourse.current_hostname
+    login_uri = if current_hostname == "localhost" then "http://localhost:3000" else "https://#{current_hostname}" end + "/auth/google_one_tap/callback"
```

Comment:
`Discourse.base_url` can take care of all the localhost/https/port logic

```suggestion
    login_uri = "#{Discourse.base_url}/auth/google_one_tap/callback"
```